### PR TITLE
#126: Bring workflow actions up to date

### LINKED ISSUE #126: Bring workflow actions up to date
## Background
In other project, namely [Atum](https://github.com/AbsaOSS/atum-service), we have better setup Workflow actions.

## Feature
Bring the action up to date:
* clear naming
* steps separation and naming
* check for test file names

### PR DESCRIPTION
**Pull request by benedeki**
* unified actions names and filenames
* added test filename check
* streamlined job steps

Closes #126

### CODE REVIEW

File: .github/workflows/test_filenames_check.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        uses: actions/checkout@v2
+
+      - name: Filename Inspector
+        id: scan-test-files
+        uses: AbsaOSS/filename-inspector@master
+        with:
+          name-patterns: '*UnitTests.*,*IntegrationTests.*'
+          paths: '**/src/test/scala/**'
+          report-format: 'console'
+          excludes: ''
```
Comment (lsulak) [inline .github/workflows/test_filenames_check.yml:39]: som exclusions are needed here as well after all

**Comment by lsulak** [comment]
What about RN check, as was done in https://github.com/AbsaOSS/atum-service/pull/212/files ?

**Comment by benedeki** [comment]
> What about RN check, as was done in https://github.com/AbsaOSS/atum-service/pull/212/files ?
> 
> actually even `release.yml` is quite old, I mean the whole pipeline and its approach

Talked to @miroslavpojer. He's heavily working a new, better solution for the release notes. He suggest not to implement the old thing on any other repo.

**Comment by benedeki** [comment]
This item depends on:
* AbsaOSS/fa-db#125
By **[Dependent Issues](https://github.com/z0al/dependent-issues)**.

File: .github/workflows/jacoco_report.yml [hunk 2]
```diff
@@ -62,8 +60,8 @@ jobs:
           paths: >
             ${{ github.workspace }}/core/target/scala-${{ env.SCALA_SHORT_VERSION }}/jacoco/report/jacoco.xml
           token: ${{ secrets.GITHUB_TOKEN }}
-          min-coverage-overall: ${{ env.COVERAGE_OVERALL_EXPECTATION }}
-          min-coverage-changed-files: ${{ env.COVERAGE_CHANGED_EXPECTATION }}
+          min-coverage-overall: ${{ 57.0 }}
```
Comment (miroslavpojer) [inline .github/workflows/jacoco_report.yml:63]: Are {{ }} still needed if inside is a number?
I believe we talked about removing these constants, right?

File: publish.sbt [hunk 3]
```diff
@@ -32,9 +32,34 @@ ThisBuild / developers := List(
...
+  ),
+  Developer(
+    id = "Zejnilovic",
+    name = "Saša Zejnilović",
+    email = "sasa.zejnilovic@absa.africa",
+    url = url("https://github.com/Zejnilovic")
   )
 )
 
+
```
Comment (miroslavpojer) [inline publish.sbt:62]: Two empty lines?

**Review by miroslavpojer — COMMENTED** [review]
- [x] code review

File: .github/workflows/jacoco_report.yml (hunk 2 above)
Comment (miroslavpojer) [reply .github/workflows/jacoco_report.yml]: Here, I pointed to 1st location where I saw it. This is on more locations in yml file.

Comment (benedeki) [reply .github/workflows/jacoco_report.yml]: Completely rewritten.

**Review by miroslavpojer — APPROVED** [review]
Last idea: how about overall values per module? There are big differences.

**Comment by benedeki** [comment]
> Last idea: how about overall values per module? There are big differences.

This is already a long workflow, with a proclivity for typo errors or omitting stuff (like modules).
A creation of dedicated action is the way to go forward.

**Pull request merged**